lib/autokey/service.py

```diff
@@ -489,28 +497,37 @@ def execute(self, script: model.Script, buffer=''):
         if script.path is not None:
             # Overwrite __file__ to contain the path to the user script instead of the path to this service.py file.
             scope["__file__"] = script.path
-        try:
-            exec(script.code, scope)
-        except Exception as e:
-            logger.exception("Script error")
-            self.error = "Script name: '{}'\n{}".format(script.description, traceback.format_exc())
-            self.app.notify_error("The script '{}' encountered an error".format(script.description))
+        self._execute(scope, script)
```

Comment:
It's confusing having such similar func names, and probably easy to make a mistake with. I'm having trouble thinking up alternatives though... `_execute_and_record_errors`?

Reply:
I can rename `execute()` to `execute_script()`. This isn’t part of any user-facing API, so a global rename is safe to do.

Reply:
Perfect. `execute_Script` or better `execute_stored_script`? to distinguish that it takes an autokey object?

Reply:
I’m opposing `execute_stored_script`. A script stored in a file pointed to by a _Path_ is technically still _stored_ (as in stored on disk). And _temporary_ AutoKey scripts (that you added support for) are technically _not_ stored (as they are in memory only). So the function `execute_stored_script` can’t execute _temporary_ Scripts?
Since that’s not true, I find that more confusing than the plain `execute_script`.